pubky-homeserver/src/data_directory/quota_config/path_limit.rs

```diff
@@ -33,20 +46,49 @@ impl PathLimit {
             quota,
             key,
             burst,
+            whitelist: vec![],
+        }
+    }
+
+    /// Check if the key is whitelisted.
+    pub fn is_whitelisted(&self, key: &LimitKey) -> bool {
+        self.whitelist.iter().any(|k| k == key)
+    }
+
+    /// Validate the path limit.
+    pub fn validate(&self) -> anyhow::Result<()> {
+        if let Some(k) = self.whitelist.iter().find(|k| k.get_type() != self.key) {
+            let should_type = self.key.to_string();
+            let is_type = k.get_type().to_string();
+            let msg = format!("Whitelist key type mismatch for '{k}'. Expected type '{should_type}' but got '{is_type}'. Full path limit: {self}");
+            return Err(anyhow::anyhow!(msg));
         }
+        Ok(())
     }
 }
 
 impl std::fmt::Display for PathLimit {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let burst_str = self
+            .burst
+            .map(|b| format!(" burst {}", b))
+            .unwrap_or("".to_string());
+        let whitelist_str = if self.whitelist.is_empty() {
+            "".to_string()
+        } else {
+            format!(
+                " whitelist: {}",
+                self.whitelist
+                    .iter()
+                    .map(|k| k.to_string())
+                    .collect::<Vec<String>>()
+                    .join(", ")
+            )
+        };
```

Comment:
NIT: IMO it's acceptable to print out debug of `Vec<LimitKey>` in this case
```suggestion
let whitelist_str = format!(" whitelist: {:?}", self.whitelist);
```

Will render like `" whitelist: [Ip(127.0.0.1)]"` instead of `" whitelist: 127.0.0.1"` and empty whitelist will look like `" whitelist: []"`. It's not as pretty, but acceptable for the one liner.